### PR TITLE
Create hdrop: emulate tdrop in Hyprland

### PR DESCRIPTION
**Pull request by Schweber**
This script allows me to emulate the main feature of tdrop in Hyprland (please see description in the script). It works for me but needs more work before it could be merged.

Please tell me if this feature is interesting for this repo. If so, i would flesh out the script and try to make it ready to merge.

### CODE REVIEW

File: hdrop/hdrop [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+CLASS="$1"
+ACTIVE_WORKSPACE="$(hyprctl activeworkspace -j | jq -r .id)"
+
+TEMP=$(getopt -q --options a: --longoptions class: -n 'hdrop' -- "$@")
+
+eval set -- "$TEMP"
+
+while true; do
+  case "$1" in
+  -a)
```
Comment (fufexan) [inline hdrop/hdrop:29]: These two can be combined as `-a|--class)`.

**Review by fufexan — COMMENTED** [review]
I think it's a good addition! Pretty simple and useful.

**Comment by fufexan** [comment]
Also, please add hdrop to the CI.
> If your utility is written in bash, please add it to the [checks matrix](https://github.com/hyprwm/contrib/blob/main/.github/workflows/check.yml).

**Comment by Schweber** [comment]
Ok, then i'll try to make the script nicer and tell you when i'm finished.

File: hdrop/hdrop (hunk 1 above)
Comment (Schweber) [reply hdrop/hdrop:29]: Thanks! I changed it.

**Comment by Schweber** [comment]
That's as far as i can take it. I adapted some parts of the other scripts.

Please take a look at `default.nix` and `Makefile`. I just adapted the one from `grimblast` and don't know how to validate them locally.

File: hdrop/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+MANDIR ?= $(DATAROOTDIR)/man
+MAN1DIR ?= $(MANDIR)/man1
+
+all: hdrop.1
+
+hdrop.1: hdrop.1.scd
+	scdoc < $< > $@
+
+install: hdrop.1 hdrop
+	@install -v -D -m 0644 hdrop.1 --target-directory "$(MAN1DIR)"
```
Comment (fufexan) [inline hdrop/Makefile:15]: Having this implies the existence of a manpage. Do you intend to write one?

Comment (Schweber) [reply hdrop/Makefile]: No, not for now. I tried to take it out, is it correct now?

Comment (fufexan) [reply hdrop/Makefile]: You can also remove any `hdrop.1` strings in there, including the entire `hdrop.1` build step.

Comment (Schweber) [reply hdrop/Makefile]: Ok i took everything out of `default.nix` and `Makefile`

File: hdrop/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        libnotify
+      ]
+      ++ lib.optional (hyprland != null) hyprland)}"
+  '';
+
+  meta = with lib; {
+    description = "Emulate 'tdrop' in Hyprland (show and hide specific programs per keybind)";
+    license = licenses.agpl;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [Schweber];
```
Comment (fufexan) [inline hdrop/default.nix:41]: This field is only for [Nixpkgs](https://github.com/NixOS/nixpkgs) maintainers. If you're not one, you can leave it empty.

Comment (Schweber) [reply hdrop/default.nix]: Ok, i set it to empty

Comment (Schweber) [reply hdrop/default.nix]: Now i commented it out. Which is better/the correct way?

Comment (fufexan) [reply hdrop/default.nix]: It's the same, since by default an unset `maintainers` means `[]` (empty list)`.

File: hdrop/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  version = "0.1";
+
+  src = ./.;
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+  ];
```
Comment (fufexan) [inline hdrop/default.nix:22]: This can also be deleted since it's empty.

**Pull request merged**